client/src/components/InputsModal/InputOAuthCredentials.tsx

```diff
@@ -64,26 +64,31 @@ const InputOAuthCredentials: FC<InputOAuthCredentialsProps> = ({
     {
       name: 'refresh_token',
       label: 'Refresh Token (will automatically refresh if available)',
+      required: false,
     },
     {
       name: 'token_url',
       label: 'Token Endpoint',
       hide: !showRefreshDetails,
+      required: true,
```

Comment:
'Hardcoding' to required: true so that the field is marked as required when it is visible.  Because if it is visible, then that means a refresh token was filled in, and the refresh token cannot be used unless this is also filled in.  Same applies to `client_id`, but not the others.